### ISSUE TITLE
Label-based indexing on scenes

### ISSUE THREAD

**Issue by gerritholl** [pytroll/satpy#1429]
## Feature Request

**Is your feature request related to a problem? Please describe.**

I have a 3-dimensional scene obtained with `sc = ms.blend(blend_function=satpy.multiscene.timeseries)`.  I would like to index the `time`-axis based on labels such as I can do with an `xarray.DataArray` using `.loc[...]` or `.sel(...)`:

```python
import satpy
import satpy.utils
satpy.utils.debug_on()
import glob

abi_dir = "/data/gholl/cache/fogtools/abi/2020/04/13/"
abi_glob = f"00/C10/OR_ABI-L1b-RadC-M6C*_G16_s2020104*_e*_c*.nc"
abi_files = glob.glob(abi_dir + abi_glob)

ms = satpy.MultiScene.from_files(
        abi_files,
        reader="abi_l1b")
ms.load(["C10", "C11"])
sc = ms.blend(blend_function=satpy.multiscene.timeseries)
idx = {"time": slice("2020-04-13T00:20", "2020-04-13T00:30")}
print(sc["C10"].loc[idx])
print(sc.loc[idx])  # not possible
```

**Describe the solution you'd like**

I would like that the `Scene` object grows support for either `.loc` or `.sel` (or both), which should return a new `Scene` object in which all datasets have been accessed using `.loc` or `.sel`, similar to how position-based indexing for `Scene` objects is already supported.

**Describe any changes to existing user workflow**

New feature without additional dependencies, no changes to user workflow.

**Additional context**

Current possible workarounds:

- Calculate the indices separately using the coordinates, use those for indexing, or
- index all datasets separately and construct a new `Scene` object using those.

Either is possible but requires additional steps and therefore additional potential bugs for the user.

**Comment by djhoese** [comment]
Initial opinion: We either change the Scene to be a subclass of xarray Dataset or we suggest users use the `.to_xarray_dataset()` method that already exists. Making a Scene a subclass of xarray Dataset has problems though for the very very common case of multiple resolutions per Scene (can't have two sets of coordinates named the same thing).

**Comment by gerritholl** [comment]
Good point, I didn't think of `to_xarray_dataset()` as a 3rd possible workaround here, but there doesn't appear to be a `from_xarray_dataset()` so if my further processing still expects a `Scene` then that workaround is incomplete.

**Comment by gerritholl** [comment]
Is there any way in which current position-based indexing of a `Scene` with multiple resolutions is at all useful (when indexing along a dimension that differs between datasets)?  I didn't think of that when writing this issue, but isn't that an even stronger argument for label-based indexing of scenes?  What does `.to_xarray_dataset()` do in this case?  On the other hand, doesn't a scene with multiple resolutions necessarily trigger the message _The following datasets were not created and may require resampling to be generated_?

**Comment by djhoese** [comment]
> Is there any way in which current position-based indexing of a Scene with multiple resolutions is at all useful (when indexing along a dimension that differs between datasets)?

I think it depends on which indexing you're talking about. I think the crop method will fail in some cases (different projections), but for different resolutions it can really help "cut off" dask chunks before doing further processing. The `crop` method at least tries to be smart about making sure the different resolution slices are aligned. Regular slicing with brackets I don't think works (errors) with multiple resolutions...I think.

>  I didn't think of that when writing this issue, but isn't that an even stronger argument for label-based indexing of scenes?

Yes, good point. I've never really gotten the hang of label based indexing because some datasets can't/shouldn't be indexed that way (at least based on my limited understanding of label based indexing). For example, data on a SwathDefinition can't guarantee the order of lon/lats. Or certain projections being indexed by lon/lat when that isn't always contiguous.

> What does .to_xarray_dataset() do in this case?

Errors out I would guess.

> On the other hand, doesn't a scene with multiple resolutions necessarily trigger the message The following datasets were not created and may require resampling to be generated?

Only if you tried to build a composite or use a modifier that needed to combine the data at different resolutions.

**Comment by gerritholl** [comment]
> > I didn't think of that when writing this issue, but isn't that an even stronger argument for label-based indexing of scenes?
> 
> Yes, good point. I've never really gotten the hang of label based indexing because some datasets can't/shouldn't be indexed that way (at least based on my limited understanding of label based indexing). For example, data on a SwathDefinition can't guarantee the order of lon/lats. Or certain projections being indexed by lon/lat when that isn't always contiguous.
> 

I think it should work as long as the coordinates are continuously strictly increasing, which should be the case for geographic coordinates on geostationary satellites.  It should also be often the case for the time coordinate in a scene constructed from a multiscene using `.blend(timeseries)` (at least in the case where the `MultiScene` is meant to represent a timeseries).

> > What does .to_xarray_dataset() do in this case?
> 
> Errors out I would guess.

I just tried it and it indeed fails with `ValueError`.

**Comment by gerritholl** [comment]
Related: #1276
